lib/SILOptimizer/Mandatory/DiagnoseInvalidEscapingCaptures.cpp

```diff
@@ -55,6 +55,16 @@ static bool checkNoEscapePartialApplyUse(Operand *oper, FollowUse followUses) {
       isa<CopyBlockWithoutEscapingInst>(user))
     return false;
 
+  // Look through copies, borrows, and conversions.
+  if (SingleValueInstruction *copy = getSingleValueCopyOrCast(user)) {
+    // Only follow the copied operand. Other operands are incidental,
+    // as in the second operand of mark_dependence.
+    if (oper->getOperandNumber() == 0)
+      followUses(copy);
+
+    return false;
+  }
+
   // Ignore uses that are totally uninteresting. partial_apply [stack] is
   // terminated by a dealloc_stack instruction.
   if (isIncidentalUse(user) || onlyAffectsRefCount(user) ||
```

Comment:
This is fragile. You should at least comment the assumption that the `getSingleValueCopyOrCast` check above handles all instructions that produce results for which `onlyAffectsRefCount` returns true. Ideally this case would have an assert that `user` produces no results.

Reply:
Added comment.

Reply:
When `onlyAffectRefCount(user)` returns true, I'm curious if `user` ever has results. And if so, how we know it's ok not to follow those results...

Reply:
Ah okay, I added the assert that you suggested.

Reply:
Ah okay, I added the assert that you suggested.